argilla/src/argilla/client.py

```diff
@@ -188,9 +206,26 @@ def __init__(self, client: "Argilla") -> None:
         self._client = client
         self._api = client.api.workspaces
 
-    def __call__(self, name: str) -> Optional["Workspace"]:
-        workspace_models = self._api.list()
+    @overload
+    def __call__(self, name: str) -> Optional["Workspace"]: ...
+
+    @overload
+    def __call__(self, id: Union[UUID, str]) -> "Workspace": ...
 
+    def __call__(self, name: str = None, id: Union[UUID, str] = None) -> Optional["Workspace"]:
+        if not (name or id):
+            raise ArgillaError("One of 'name' or 'id' must be provided")
+
+        if name and id:
+            warnings.warn("Only one of 'name' or 'id' must be provided. Using 'id'")
+            name = None
+
+        if id is not None:
+            if not isinstance(id, UUID):
+                id = UUID(id)
+            return self._from_model(self._api.get(id))
```

Comment:
should we provide a warning if not found? Or perhaps use similar logic as with the `name`